### PR TITLE
Merge 2.6 into develop 

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

This is a merge of 2.6 branch into develop.

## QA steps

Ensure version is correct for the develop branch - `2.7-beta1`

### CODE REVIEW

File: acceptancetests/jujupy/client.py [hunk 1]
```diff
@@ -1391,6 +1462,9 @@ def _get_substrate_constraints(self):
             # For now only maas support spaces in a meaningful way.
             return 'mem=2G spaces={}'.format(','.join(
                 '^' + space for space in sorted(self.excluded_spaces)))
+        elif self.env.lxd:
+            # LXD should not me constrained via memory
```
Comment (manadart) [inline acceptancetests/jujupy/client.py]: "be"

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
> FAIL: flag_test.go:94: WorkerSuite.TestErrRefresh
> https://pastebin.canonical.com/p/cf5tNDm6qg/

$$merge$$

**Pull request merged**